File: tests/expected/infer_ops.kt


fun foo() {
    val a = 10
    val b = 20
    val _c1 = (a + b)
    val _c2 = (a - b)
    val _c3 = (a * b)
    val _c4 = (a / b)
    val _c5 = -(a)
    val d = 2.0
    val _e1 = (a.toDouble() + d)
    val _e2 = (a.toDouble() - d)
    val _e3 = (a.toDouble() * d)
    val _e4 = (a.toDouble() / d)
    val _f = -3.0
    val _g = -(a)
}

fun add1(
    x: Byte,
    y: Byte,
): Short = (x + y) as Short

fun add2(
    x: Short,
    y: Short,
): Int = (x + y)

fun add3(
    x: Int,
    y: Int,
): Long = (x + y) as Long

fun add4(
    x: Long,
    y: Long,
): Long = (x + y)

fun add5(
    x: UByte,
    y: UByte,
): UShort = (x + y) as UShort

fun add6(
    x: UShort,
    y: UShort,
): UInt = (x + y)

fun add7(
    x: UInt,
    y: UInt,
): ULong = (x + y) as ULong

fun add8(
    x: ULong,
    y: ULong,
): ULong = (x + y)

fun add9(
    x: Byte,
    y: UShort,
): UInt = (x.toUShort() + y)

fun sub(
    x: Byte,
    y: Byte,
): Byte = (x - y) as Byte

fun mul(
    x: Byte,
    y: Byte,
): Short = (x * y) as Short

fun fadd1(
    x: Byte,
    y: Double,
): Double = (x.toDouble() + y)

fun show() {
    assert(fadd1(6, 6.0) == 12.toDouble())
    println("OK")
}

fun main(argv: Array<String>) {
    foo()
    show()
}

File: tests/expected/infer_ops.dart
// @dart=3.4
import 'package:sprintf/sprintf.dart';

foo() {
  final int a = 10;
  final int b = 20;
  final int _c1 = (a + b);
  final int _c2 = (a - b);
  final int _c3 = (a * b);
  final int _c4 = (a ~/ b);
  final int _c5 = -(a);
  final double d = 2.0;
  final double _e1 = (a + d);
  final double _e2 = (a - d);
  final double _e3 = (a * d);
  final double _e4 = (a / d);
  final double _f = -3.0;
  final int _g = -(a);
}

int add1(int x, int y) {
  return (x + y);
}

int add2(int x, int y) {
  return (x + y);
}

int add3(int x, int y) {
  return (x + y);
}

int add4(int x, int y) {
  return (x + y);
}

int add5(int x, int y) {
  return (x + y);
}

int add6(int x, int y) {
  return (x + y);
}

int add7(int x, int y) {
  return (x + y);
}

int add8(int x, int y) {
  return (x + y);
}

int add9(int x, int y) {
  return (x + y);
}

int sub(int x, int y) {
  return (x - y);
}

int mul(int x, int y) {
  return (x * y);
}

double fadd1(int x, double y) {
  return (x + y);
}

show() {
  assert(fadd1(6, 6.0) == 12);
  print(sprintf("%s", ["OK"]));
}

main(List<String> argv) {
  foo();
  show();
}
